app/infrastructure/persistence/models/permission.py

```
@@ -0,0 +1,69 @@
+    resource: Mapped[str] = mapped_column(String, nullable=False, index=True)
+    action: Mapped[str] = mapped_column(String, nullable=False, index=True)
+    description: Mapped[str | None] = mapped_column(Text, nullable=True)
+
+    __table_args__ = (
+        UniqueConstraint("tenant_id", "code", name="uq_permission_tenant_code"),
+        Index("ix_permission_resource_action", "tenant_id", "resource", "action"),
+    )
+
+
+class RolePermission(CuidMixin, TenantMixin, Base):
+    """Many-to-many role–permission. Table: role_permission."""
```

Ruff (0.14.14)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)